fineract-provider/src/main/java/org/apache/fineract/portfolio/client/domain/AccountNumberGenerator.java

```diff
@@ -108,7 +124,14 @@ private String generateAccountNumber(Map<String, String> propertyMap, AccountNum
             }
         }
 
-        String accountNumber = StringUtils.leftPad(propertyMap.get(ID), accountMaxLength, '0');
+        final GlobalConfigurationPropertyData randomAccountNumber = this.configurationReadPlatformService
+                .retrieveGlobalConfiguration("random-account-number");
```

Comment:
Does it make sense to have this as a single global configuration property so that then all numbers are random (client, loan, savings account)? Or would it make more sense to store this as part of the AccountNumberFormat as a parameter so that people can enable this for different entities if they'd like to? Then we could also store the length of the random number to be created as part of the AccountNumberFormat.